pkg/resources/manifests.go

```diff
@@ -291,9 +295,25 @@ func GenerateDeploymentManifest(ctx context.Context, workspaceObj *kaitov1alpha1
 					Controller: &controller,
 				},
 			},
+			Annotations: map[string]string{
+				WorkspaceRevisionAnnotation: workloadHash,
+			},
 		},
 		Spec: appsv1.DeploymentSpec{
 			Replicas: lo.ToPtr(int32(replicas)),
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RollingUpdateDeploymentStrategyType,
+				RollingUpdate: &appsv1.RollingUpdateDeployment{
+					MaxSurge: &intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 0,
+					},
+					MaxUnavailable: &intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 1,
+					},
```

Comment:
Can you explain the MaxSurge 0 and MaxUnavailable: 1? Provide comment. I know you discussed with Fei - trying to catch up